src/realm/database_element.hpp

```diff
@@ -53,11 +64,50 @@ class ArrayParent {
     // Used only by Array::to_dot().
     virtual std::pair<ref_type, size_t> get_to_dot_parent(size_t ndx_in_parent) const = 0;
 
-    friend class DbElement;
+    friend class DatabaseElement;
     friend class Array;
 };
 
-class DbElement {
+/// Provides access to individual nodes of the database.
+///
+/// This class serves purely as an accessor, it assumes no ownership of the
+/// referenced memory.
+///
+/// A database node accessor can be in one of two states: attached or unattached.
```

Comment:
The comments seems to suggest this should be called `NodeAccessor` or something like that?